freeipa/src/main/java/com/sequenceiq/freeipa/service/binduser/UserSyncBindUserService.java

```diff
@@ -0,0 +1,96 @@
+
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.freeipa.api.v1.ldap.model.describe.DescribeLdapConfigResponse;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.FreeIpaClientExceptionUtil;
+import com.sequenceiq.freeipa.client.model.User;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.ldap.LdapConfig;
+import com.sequenceiq.freeipa.ldap.LdapConfigService;
+import com.sequenceiq.freeipa.ldap.v1.LdapConfigV1Service;
+import com.sequenceiq.freeipa.service.freeipa.FreeIpaClientFactory;
+
+@Service
+public class UserSyncBindUserService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserSyncBindUserService.class);
+
+    private static final String USERSYNC_USER_POSTFIX = "usersync";
+
+    private static final String USER_ADMINISTRATOR_ROLE = "User Administrator";
+
+    @Inject
+    private LdapConfigV1Service ldapConfigV1Service;
+
+    @Inject
+    private LdapConfigService ldapConfigService;
+
+    @Inject
+    private LdapBindUserNameProvider userNameProvider;
+
+    @Inject
+    private FreeIpaClientFactory ipaClientFactory;
+
+    public void createUserAndLdapConfig(Stack stack) throws FreeIpaClientException {
+        cleanupAlreadyExistingLdapConfig(stack.getEnvironmentCrn(), stack.getAccountId());
+        ldapConfigV1Service.createNewLdapConfig(stack.getEnvironmentCrn(), USERSYNC_USER_POSTFIX, stack, true);
+        addUserAdminRoleForBindUser(stack);
+    }
+
+    public boolean doesBindUserAndConfigAlreadyExist(Stack stack) throws FreeIpaClientException {
+        String bindUserName = getUserSyncBindUserName();
+        Optional<LdapConfig> ldapConfig =
+                ldapConfigService.find(stack.getEnvironmentCrn(), stack.getAccountId(), bindUserName);
+        if (ldapConfig.isPresent()) {
+            LOGGER.info("LdapConfig for usersync bind user exists");
+            FreeIpaClient client = ipaClientFactory.getFreeIpaClientForStack(stack);
+            Optional<User> user = client.userFind(bindUserName);
+            LOGGER.info("Bind user in FreeIPA {}", user.isPresent() ? "exists" : "doesn't exist");
+            return user.isPresent();
+        } else {
+            LOGGER.info("LdapConfig for usersync bind user doesn't exist");
+            return false;
+        }
+    }
+
+    private String getUserSyncBindUserName() {
+        return userNameProvider.createBindUserName(USERSYNC_USER_POSTFIX);
+    }
+
+    public DescribeLdapConfigResponse getUserSyncLdapConfigIfExistsOrThrowNotFound(String environmentCrn, String accountId) {
+        Optional<LdapConfig> ldapConfig =
+                ldapConfigService.find(environmentCrn, accountId, getUserSyncBindUserName());
+        if (ldapConfig.isPresent()) {
+            return ldapConfigV1Service.convertLdapConfigToDescribeLdapConfigResponse(ldapConfig.get());
```

Comment:
I saw that this method was changed from private to public in the `LdapConfigV1Service`. This feels strange to me to call into another business logic class to do model conversions. I know that there are `Converter`s elsewhere in the codebase. Would it make sense to pull this method out into a `Converter` or otherwise separate out the model conversion methods into a separate class?

Reply:
I can move it out to a converter